src/cache-utils.ts

```diff
@@ -47,11 +51,24 @@ export const getCommandOutput = async (toolCommand: string) => {
   return stdout.trim();
 };
 
-const getPackageManagerVersion = async (
+export const getPackageManagerWorkingDir = (): string | null => {
+  const cache = core.getInput('cache');
+  if (cache !== 'yarn') {
+    return null;
+  }
+
+  const cacheDependencyPath = core.getInput('cache-dependency-path');
+  return cacheDependencyPath ? path.dirname(cacheDependencyPath) : null;
+};
+
+export const getPackageManagerVersion = async (
   packageManager: string,
   command: string
 ) => {
-  const stdOut = await getCommandOutput(`${packageManager} ${command}`);
+  const stdOut = await getCommandOutput(
+    `${packageManager} ${command}`,
+    getPackageManagerWorkingDir()
```

Comment:
I think we can call `getPackageManagerWorkingDir` directly in getCommandOutput.

Reply:
it would happen a pain some day to use `getCommandOutput` for other purposes. [I wrapped it](https://github.com/actions/setup-node/pull/735/commits/352719599292b2ba43c5e7a43e354ac8c2cdf395#diff-b602b70ff15dc447863bdb26a26be6c2ef6c8aa51314213a26f1e0f162618abfR64`) into `getPackageManagerCommandOutput` instead